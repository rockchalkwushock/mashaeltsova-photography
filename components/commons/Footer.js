import { styles } from '../../lib'

const Footer = ({ children }) => (
  <footer>
    {children}
    <style jsx>{`
      footer {
        background-color: ${styles.colors.peach};
        display: flex;
        flex-direction: column;
        padding: 1em;
      }
      @media (min-width: 768px) {
        footer {
          align-items: center;
          flex-direction: row;
        }
      }
    `}</style>
  </footer>
)

export default Footer
