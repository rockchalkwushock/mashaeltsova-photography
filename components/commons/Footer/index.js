/**
 * REVIEW
 * FIXME
 * I will need to pass props on this like `intl`
 */
const Footer = ({ children }) => (
  <footer>
    {children}
    <style>{`
    footer {
      background-color: lightcoral;
      grid-column: span 7;
      height: 50px;
    }
  `}</style>
  </footer>
)

export default Footer
