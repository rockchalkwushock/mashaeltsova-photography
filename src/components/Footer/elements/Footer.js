import styled from 'styled-components'

const Footer = styled.footer`
  background-color: #333;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-top: 2em;
  text-align: center;
  width: 100vw;
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: 2em 1em 1em 1em;
  }
`

Footer.displayName = 'Footer'

export default Footer
