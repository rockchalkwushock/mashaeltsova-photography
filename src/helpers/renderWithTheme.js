import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
