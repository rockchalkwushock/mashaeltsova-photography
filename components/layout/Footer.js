import { Container, Icon } from '../commons'
import { styles } from '../../lib'

const Footer = () => (
  <footer>
    <Container className="social">
      <Icon
        icon="facebook"
        size="2x"
        url="https://www.facebook.com/masha.eltsova"
      />
      <Icon
        icon="instagram"
        size="2x"
        url="https://www.instagram.com/mashaeltcovaphotography"
      />
      <Icon icon="vk" size="2x" url="https://www.vk.com/club65938200" />
      <Icon icon="twitter" size="2x" url="https://twitter.com/MASHAELTSOVA" />
    </Container>
    <Container className="copyright">
      <Icon icon="copyright" size="lg" url="https://codybrunner.rocks" />
    </Container>
    <style jsx>{`
      footer {
        background-color: ${styles.colors.peach};
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
        padding: 1em;
      }
      @media (min-width: 768px) {
        footer {
          -webkit-box-align: center;
          -ms-flex-align: center;
          align-items: center;
          -webkit-box-orient: horizontal;
          -webkit-box-direction: normal;
          -ms-flex-direction: row;
          flex-direction: row;
        }
      }
    `}</style>
  </footer>
)

export default Footer
