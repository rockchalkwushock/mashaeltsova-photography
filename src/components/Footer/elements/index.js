import Footer from './Footer'
import FooterContainer from './FooterContainer'

export { Footer, FooterContainer }
