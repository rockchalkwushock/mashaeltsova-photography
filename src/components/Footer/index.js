import React from 'react'

import { Footer, FooterContainer } from './elements'

const AppFooter = () => (
  <Footer>
    <FooterContainer left>
      <h2>Left</h2>
    </FooterContainer>
    <FooterContainer>
      <h2>Right</h2>
    </FooterContainer>
  </Footer>
)

export default AppFooter
