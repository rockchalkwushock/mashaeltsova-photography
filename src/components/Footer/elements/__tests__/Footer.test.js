import React from 'react'

import { renderWithTheme } from '../../../../helpers'
import Footer from '../Footer'

describe('Styled Component: <Footer />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(<Footer />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('background-color', '#333')
    expect(wrapper).toHaveStyleRule('display', 'flex')
    expect(wrapper).toHaveStyleRule('flex-direction', 'column')
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('padding-top', '2em')
    expect(wrapper).toHaveStyleRule('text-align', 'center')
    expect(wrapper).toHaveStyleRule('width', '100vw')
    expect(wrapper).toHaveStyleRule('align-items', 'center', {
      media: '(min-width:736px)'
    })
    expect(wrapper).toHaveStyleRule('flex-direction', 'row', {
      media: '(min-width:736px)'
    })
    expect(wrapper).toHaveStyleRule('justify-content', 'space-between', {
      media: '(min-width:736px)'
    })
    expect(wrapper).toHaveStyleRule('padding', '2em 1em 1em 1em', {
      media: '(min-width:736px)'
    })
  })
})
