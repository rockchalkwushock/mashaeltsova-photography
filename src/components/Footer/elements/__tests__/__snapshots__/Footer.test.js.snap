// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <Footer /> should render markup 1`] = `
.c0 {
  background-color: #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 2em;
  text-align: center;
  width: 100vw;
}

@media (min-width:736px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 2em 1em 1em 1em;
  }
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Footer>
    <footer
      className="c0"
    />
  </Footer>
</ThemeProvider>
`;
